### PR TITLE
DOCS: Document suboverflowed backends extension, view, and monitoring for overflowed backends

### PR DESCRIPTION
**Pull request by VMWareJennyBeth**

### CODE REVIEW

**Review by zjpedu — COMMENTED** [review]
I will review this pr.

```
Greenplum Database includes an extension -- `gp_subtransaction_overflow` -- and a view that's -- `gp_suboverflowed_backend` -- that's run over a user-defined function to help users query for suboverflowed backends. Users can use segment id and ~and~ process id information reported in the view to terminate the offending backends, thereby preventing degradation of performance.
```

File: gpdb-doc/markdown/admin_guide/managing/monitor.html.md [hunk 1]
```diff
@@ -197,6 +198,76 @@ This occurs when the input to a hash join operator is skewed. It does not preven
...
+    -1 | 
+     0 |
+     1 | 
+     2 | 
+   (4 rows)
+    ```
+
 ## <a id="topic24"></a>Viewing Metadata Information about Database Objects 
 
 Greenplum Database tracks various metadata information in its system catalogs about the objects stored in a database, such as tables, views, indexes and so on, as well as global objects such as roles and tablespaces.
```
Comment (zjpedu) [inline gpdb-doc/markdown/admin_guide/managing/monitor.html.md:273]: , and so on,

File: gpdb-doc/markdown/admin_guide/managing/monitor.html.md [hunk 2]
```diff
@@ -197,6 +198,76 @@ This occurs when the input to a hash join operator is skewed. It does not preven
...
 5.  Check the cardinality of the join keys. If they have low cardinality, try to rewrite the query with different joining columns or additional filters on the tables to reduce the number of rows. These changes could change the query semantics.
 
+## <a id="overflowed_backends"></a>Checking for and Terminating Overflowed Backends
+
+Subtransaction overflow arises when a Greenplum Database backend creates more than 64 subtransactions, resulting in a high lookup cost. This slows query performance, but even more so when it occurs in combination with long running transactions, which result in still more lookups. Terminating suboverflowed backends and/or backends with long running transactions can help prevent and alleviate performance problems. 
+
+Greenplum Database includes an extension -- `gp_subtransaction_overflow` -- and a view that's -- `gp_suboverflowed_backend` -- that's run over a user-defined function to help users query for suboverflowed backends. Users can use segment id and and process id information reported in the view to terminate the offending backends, thereby preventing degradation of performance.
```
Comment (zjpedu) [inline gpdb-doc/markdown/admin_guide/managing/monitor.html.md:205]: segment id and ~and~

File: gpdb-doc/markdown/ref_guide/modules/intro.html.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ The following Greenplum Database and PostgreSQL `contrib` modules are installed;
 -   [fuzzystrmatch](fuzzystrmatch.html) - Determines similarities and differences between strings.
 -   [gp\_legacy\_string\_agg](gp_legacy_string_agg.html) - Implements a legacy, single-argument `string_agg()` aggregate function that was present in Greenplum Database 5.
 -   [gp\_sparse\_vector](gp_sparse_vector.html) - Implements a Greenplum Database data type that uses compressed storage of zeros to make vector computations on floating point numbers faster.
+-   [gp_subtransaction_overflow](gp_subtransaction_overflow.html) - Provides a view and user defined function for querying for suboverflowed backends.
```
Comment (zjpedu) [inline gpdb-doc/markdown/ref_guide/modules/intro.html.md:18]: user-defined is better?

File: gpdb-doc/markdown/ref_guide/system_catalogs/gp_suboverflowed_backend.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# gp_suboverflowed_backend 
+
+The `gp_suboveflowed_backend` view allows administrators to identify sessions in which a backend has subtransaction overflows coupled 
+with long running queries -- including queries on catalog tables -- which can cause query performance degradation in the system.
```
Comment (zjpedu) [inline gpdb-doc/markdown/ref_guide/system_catalogs/gp_suboverflowed_backend.html:4]: long-running is commonly seen.

File: gpdb-doc/markdown/ref_guide/modules/gp_subtransaction_overflow.html.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# gp_subtransaction_overflow
+
+The `gp_subtransaction_overflow` module implements a Greenplum Database view and user defined function for querying for backends experiencing
```
Comment (zjpedu) [inline gpdb-doc/markdown/ref_guide/modules/gp_subtransaction_overflow.html.md:3]: user-defined?

**Review by zjpedu — APPROVED** [review]
LGTM. But there are some small suggestions for improvement.

File: gpdb-doc/markdown/admin_guide/managing/monitor.html.md (hunk 1 above)
Comment (VMWareJennyBeth) [reply gpdb-doc/markdown/admin_guide/managing/monitor.html.md:273]: \

> I will review this pr.
> 
> But before that I have a confusion, why are these changes on the master branch?
> 
> In fact, the current master branch implementation has not been decided yet. I just made changes on 6x_stable.

On the Greenplum docs team, we work in master for our 6X updates. We then cherry-pick into 6X_STABLE. That is our process.

File: gpdb-doc/markdown/admin_guide/managing/monitor.html.md [hunk 6]
```diff
@@ -197,6 +198,76 @@ This occurs when the input to a hash join operator is skewed. It does not preven
 4.  Check whether the filters applied in the query match distribution keys of the base tables. If the filters and distribution keys are the same, consider redistributing some of the base tables with different distribution keys.
 5.  Check the cardinality of the join keys. If they have low cardinality, try to rewrite the query with different joining columns or additional filters on the tables to reduce the number of rows. These changes could change the query semantics.
 
+## <a id="overflowed_backends"></a>Checking for and Terminating Overflowed Backends
+
+Subtransaction overflow arises when a Greenplum Database backend creates more than 64 subtransactions, resulting in a high lookup cost. This slows query performance, but even more so when it occurs in combination with long running transactions, which result in still more lookups. Terminating suboverflowed backends and/or backends with long running transactions can help prevent and alleviate performance problems.
```
Comment (soumyadeep2007) [inline gpdb-doc/markdown/admin_guide/managing/monitor.html.md:203]: > resulting in a high lookup cost

resulting in a high lookup cost for visibility checks

Just to make it more specific?

File: gpdb-doc/markdown/ref_guide/modules/gp_subtransaction_overflow.html.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# gp_subtransaction_overflow
+
+The `gp_subtransaction_overflow` module implements a Greenplum Database view and user defined function for querying for backends experiencing 
+subtransaction overflow; these are backends that have created more than 64 subtransactions, resulting in a high lookup cost.
```
Comment (soumyadeep2007) [inline gpdb-doc/markdown/ref_guide/modules/gp_subtransaction_overflow.html.md:4]: high lookup cost for visibility checks

File: gpdb-doc/markdown/ref_guide/system_catalogs/gp_suboverflowed_backend.html (hunk 4 above)
Comment (soumyadeep2007) [inline gpdb-doc/markdown/ref_guide/system_catalogs/gp_suboverflowed_backend.html:4]: It doesn't actually detect long running queries.

> The `gp_suboveflowed_backend` view allows administrators to identify sessions in which a backend has subtransaction overflows coupled 
with long running queries -- including queries on catalog tables -- which can cause query performance degradation in the system.  

The `gp_suboveflowed_backend` view allows administrators to identify sessions in which a backend has subtransaction overflows, which can cause query performance degradation in the system, including catalog queries.

**Review by soumyadeep2007 — APPROVED** [review]
LGTM! This topic will have to be updated for 7X as there will be no extension in 7X etc, as we discussed.

**Pull request closed without merging**